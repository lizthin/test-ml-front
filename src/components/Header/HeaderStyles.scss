@import "../../assets/styles/colors.scss";
@import "../../assets/styles/mixins.scss";

.nav-header {
    height: 54px;
    width: 100%;
}

.nav-header {
    background-color: $yellow;
    font-size: 13px;
    font-weight: 400;
}

.nav-header,
.nav-header *,
.nav-header :after,
.nav-header :before {
    -webkit-box-sizing: border-box;
    box-sizing: border-box;
}

.nav-header {
    font-family: "Proxima Nova", -apple-system, "Helvetica Neue", Helvetica, Roboto, Arial, sans-serif;
    color: $black;
    -ms-user-select: none;
    user-select: none;
    -webkit-tap-highlight-color: $black;
    -webkit-tap-highlight-color: transparent;
    cursor: default;
    display: flex;
    -ms-flex-wrap: wrap;
    flex-wrap: wrap;
}

.nav-header:before {
    position: absolute;
    width: 100%;
    height: 54px;
    left: 0;
    top: 0;
}

.nav-logo-container {
    height: 36px;
    width: 53px;
    float: left;
}

.nav-logo {
    background-image: url("../../assets/img/header/Logo_ML@2x.png.png.png");
    background-size: 100% 100%;
    background-position: 100% 100%;
    background-repeat: no-repeat;
    width: 100%;
    height: 100%;
    overflow: hidden;
}

.nav-logo {
    display: inline-block;
}

.nav-input-container {
    height: 38px;
    width: 100%;
    float: left;
    display: flex;
}

.nav-form {
    width: 100%;
}

.nav-input {
    width: 95%;
    height: 100%;
    border: 1px solid $gray-light;
    border-radius: 3px;
    font-size: 18px;
    padding-left: 15px;
    color: $black;
    float: left;
    margin-left: -4px;
    @include for-phone-only {
        width: 85%;
    }
    @include for-tablet-portrait-up {
       // width: 85%;
    }
}

.nav-input::placeholder {
    color: $gray-dark;
    opacity: 1;
}

.nav-input:-ms-input-placeholder {
    color: $gray-dark;
}

.nav-input::-ms-input-placeholder {
    color: $gray-dark;
}

.nav-button {
    background-image: url("../../assets/img/header/ic_Search@2x.png.png.png");
    background-size: 50% 50%;
    background-position: 50% 50%;
    background-repeat: no-repeat;
    width: 5%;
    height: 100%;
    float: left;
    border: 1px solid $gray-light;
    @include for-phone-only {
        background-image: url("../../assets/img/header/ic_Search.png");
        background-size: 30% 50%;
        width: 15%;
    }
    background-color: #EEEEEE;
}
.container-general {
    width: 100%;
    /* padding-right: 15px; */
    /* padding-left: 15px; */
    margin-right: auto;
    margin-left: auto;
}
