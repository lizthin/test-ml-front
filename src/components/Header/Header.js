import React, { useState, useEffect } from "react";
import PropTypes from 'prop-types';
import "./HeaderStyles.scss";

const Header = ({ parentHistory }) => {
  const [query, setQuery] = useState("");

  useEffect(() => {
    if (parentHistory.location.search !== "") {
      const newQuery = parentHistory.location.search.split("=")[1];
      setQuery(newQuery);
    }
  }, [parentHistory]);

  const handleSubmitSearch = (e) => {
    e.preventDefault();
    if (query.trim().length === 0) return;
    parentHistory.push({
      pathname: "/items",
      search: `?search=${query.trim()}`,
    });
  };

  return (
    <header className="nav-header">
      <div className="container-general">
        <div className="row justify-content-center mt-2">
          <div className="col-11 col-lg-10">
            <div className="row justify-content-between">
              <div className="col-2 col-lg-1">
                <div className="nav-logo-container">
                  <a className="nav-logo"></a>
                </div>
              </div>
              <div className="col-10 col-lg-11">
                <div className="nav-input-container">
                  <form onSubmit={handleSubmitSearch} className="nav-form">
                    <input
                      type="text"
                      className="nav-input"
                      value={query}
                      onChange={(e) => setQuery(e.target.value)}
                      placeholder="Nunca dejes de buscar"
                    />
                    <button className="nav-button" />
                  </form>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </header>
  );
};

Header.propTypes = {
    parentHistory: PropTypes.object.isRequired,
}
export default Header;
