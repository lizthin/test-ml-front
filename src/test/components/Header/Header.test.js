import React from "react";
import { shallow, mount } from "enzyme";
import Header from "../../../components/Header/Header";
import { act } from "react-dom/test-utils";

describe("should create header whit input", () => {
  
  const mockProperties = {
    location: {
      hash: "",
      key: "6ti0yy",
      pathname: "/items",
      search: "?search=futbol",
    },
  };

  test("should create header", () => {
    const wrapper = shallow(<Header parentHistory={mockProperties} />);
    expect(wrapper).toMatchSnapshot()
  });

  // test('should useEffect', () => {
  //   const event = { 
  //     defaultPrevented: true,
  //     preventDefault: () => {} };
  //   jest.spyOn(event, 'preventDefault');
    
  //   const wrapper = mount(<Header parentHistory={mockProperties} />);
  //     wrapper.find('.nav-form').simulate('submit')(event);

  //   expect(wrapper).toMatchSnapshot();
    
  // })
  
});
