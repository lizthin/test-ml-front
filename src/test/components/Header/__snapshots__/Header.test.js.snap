// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should create header whit input should create header 1`] = `
<header
  className="nav-header"
>
  <div
    className="container-general"
  >
    <div
      className="row justify-content-center mt-2"
    >
      <div
        className="col-11 col-lg-10"
      >
        <div
          className="row justify-content-between"
        >
          <div
            className="col-2 col-lg-1"
          >
            <div
              className="nav-logo-container"
            >
              <a
                className="nav-logo"
              />
            </div>
          </div>
          <div
            className="col-10 col-lg-11"
          >
            <div
              className="nav-input-container"
            >
              <form
                className="nav-form"
                onSubmit={[Function]}
              >
                <input
                  className="nav-input"
                  onChange={[Function]}
                  placeholder="Nunca dejes de buscar"
                  type="text"
                  value=""
                />
                <button
                  className="nav-button"
                />
              </form>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</header>
`;
